verification/envs/python_3/main.py

```diff
@@ -47,63 +32,67 @@ class PlayerRefereeClient(RefereeClient):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._events = {}
+        self.events_call = Queue()
         self.runner = None
 
     def set_runner(self, runner):
         self.runner = runner
 
-    def _get_response_json(self):
-        response = super()._get_response_json()
-        if not response:
-            return
+    def _request(self, data, skipp_result=None):
+        data['status'] = 'success'
+        return self.request(data, skipp_result, skip_clean_up=True)
 
-        method = response.get('method')
-        if not method:
-            return response
+    def request(self, *args, **kwargs):
+        skip_clean_up = kwargs.pop('skip_clean_up', None)
 
-        if method == 'event':
-            # In situation, when user code select data but received event,
-            # we mut finish select and then run event.
-            # It's can be multiple event calls per one select, so all events put into FIFO
-            lookup_key, data = response['lookup_key'], response['data']
-            send_event = partial(self._send_event, lookup_key=lookup_key, data=data)
-            EVENTS_CALLS.put(send_event)
-            if not self.runner.is_executing:
-                run_events()
-            return self._get_response_json()
-        return response
+        if not skip_clean_up:
+            self.clean_up()
 
-    def _request(self, data, skipp_result=None):
-        data['status'] = 'success'
-        return self.request(data, skipp_result)
+        return super().request(*args, **kwargs)
+
+    def clean_up(self):
+        while not self.events_call.empty():
+            response = self.events_call.get()
+            self.runner.action_event(response)
 
     def _send_event(self, lookup_key, data):
         callback = self._events[lookup_key]
         callback(data=data)
 
     def _subscribe(self, lookup_key, callback):
-        self._events[lookup_key] = callback
+        self.runner.subscribe(lookup_key, callback)
+
+    def wait_actual_response(self, response):
+        if response.get('action') != 'event':
+            return response
+
+        self.events_call.put(response)
+        return self.wait_actual_response(self._get_response_json())
+
+    def actual_request(self, *args, **kwargs):
```

Comment:
where is 3 methods for request: `request`, `_request`, `actual_request`.

Can we override man `request` and put all code to it?